### ISSUE TITLE
Guard coding style with autoformatter

### ISSUE THREAD

**Issue by raboof** [raboof/notion#125]
The Notion coding style is described at http://notion.sourceforge.net/notionnotes/node7.html .

Unfortunately, it is not applied consistently, and a bit unorthodox.

It would be good to follow a consistent coding style, but I think it should not be manually enforced: we should just have an autoformatter.

This might be a good opportunity to also switch to a more conventional coding style.

I've had reasonably good experience with clang-format (with the default settings) in other projects, perhaps we should adopt that/those?

**Comment by wilhelmy** [comment]
The problem with adopting a new coding style (at least if we convert everything that already exists) is that we lose `git blame`-ability. So I'm in favour of using something more widely known for new files (and maybe eben on a function level), but not for a global autoformat run.

**Comment by raboof** [comment]
> we lose git blame-ability

I don't think that is a big problem: it may be a bit less convenient, but the [GitHub 'Git Blame' functionality](https://github.com/raboof/notion/blame/master/notion/notion.c) has the 'View blame prior to this change' buttons feature which makes this pretty easy to navigate. I suspect there's git clients who can do this as well, though I don't use those myself.

> So I'm in favour of using something more widely known for new files (and maybe eben on a function level), but not for a global autoformat run

I think it is better to be consistent.

When we reformatted https://github.com/akka/akka the main nuisance was that it (as expected) created conflicts for in-flight PR's. The fact that 'git blame' became a bit harder hasn't been a problem.

**Comment by wilhelmy** [comment]
After 4.0, then, when we finished off all open PRs?

**Comment by raboof** [comment]
> After 4.0, then, when we finished off all open PRs?

Sure, that could be good timing! I don't like introducing dependencies between tasks, so:

* I volunteer to fix any merge conflicts whenever this issue has been picked up and merged.
* I think this issue can be, but doesn't have to be, picked up before cutting Notion 4.
